src/util/entitlements/projects.tsx

```diff
@@ -0,0 +1,38 @@
+import { useAuth } from "context/auth";
+import { hasEntitlement } from "./helpers";
+import { LxdProject } from "types/project";
+import { useProject } from "context/project";
+
+export const useProjectEntitlements = (project?: LxdProject) => {
+  const { isFineGrained } = useAuth();
+  const { project: currentProject } = useProject();
+
+  const validProject = project || currentProject;
```

Comment:
We should not have this other way of fetching the current project in here. Better keep the interface consistent and expect a project to be passed in. Callers then would need to fetch the current project.